tools/sync.js

```diff
@@ -0,0 +1,214 @@
+/*
+Name: Ethereum Blockchain syncer
+Version: .0.0.1
+This file will start syncing the blockchain from the node address you provide in the conf.json file.
+Please read the README in the root directory that explains the parameters of this code
+*/
+require( '../db.js' );
+var etherUnits = require("../lib/etherUnits.js");
+var BigNumber = require('bignumber.js');
+
+var fs = require('fs');
+var Web3 = require('web3');
+
+var mongoose        = require( 'mongoose' );
+var Block           = mongoose.model( 'Block' );
+var Transaction     = mongoose.model( 'Transaction' );
+
+//Just listen for latest blocks and sync from the start of the app.
+var listenBlocks = function(config) {
+    var newBlocks = web3.eth.filter("latest");
+    newBlocks.watch(function (error, log) {
+    if(error) {
+        console.log('Error: ' + error);
+    } else if (log == null) {
+        console.log('Warning: null block hash');
+    } else {
+      console.log('Found new block: ' + log);
+      grabBlock(config,web3,log);
+      updatedEndBlock(config,log);
+    }
+  });
+}
+//Grab latest block info and it transactions and write to db
+var grabBlock = function(config, web3, blockHashOrNumber) {
+  if(blockHashOrNumber == undefined) {
+    blockHashOrNumber = config.endBlock
+  }
+  if(web3.isConnected()) {
+    web3.eth.getBlock(blockHashOrNumber, true, function(error, blockData) {
+      if(error) {
+        console.log('Warning: error on getting block with hash/number: ' +   blockHashOrNumber + ': ' + error);
+      }else if(blockData == null) {
+        console.log('Warning: null block data received from the block with hash/number: ' + blockHashOrNumber);
+      }else{
+        writeBlockToDB(config, blockData);
+        writeTransactionsToDB(config, blockData);
+        return;
+      }
+    });
+  }else{
+    console.log('Error: Web3 connection time out trying to get block ' + blockHashOrNumber + ' retrying connection now');
+    return;
+  }
+}
+//Full chain syncer
+var grabBlock2 = function(config, web3, blockHashOrNumber) {
+  if(blockHashOrNumber == undefined) {
+    blockHashOrNumber = config.endBlock
+  }
+  if(web3.isConnected()) {
+    web3.eth.getBlock(blockHashOrNumber, true, function(error, blockData) {
+      if(error) {
+        console.log('Warning: error on getting block with hash/number: ' +   blockHashOrNumber + ': ' + error);
+      }else if(blockData == null) {
+        console.log('Warning: null block data received from the block with hash/number: ' + blockHashOrNumber);
+       }else{
+        if(config.lastSynced === 0){
+          writeBlockToDB(config, blockData);
```

Comment:
`writeBlockToDB`, `writeTransactionsToDB`, and `updateLastSynced` don't need to be repeated, both the `if` and `else` accomplish largely the same thing.